app/lib/shared/storage.dart

```diff
@@ -31,13 +32,70 @@ Future<Bucket> getOrCreateBucket(Storage storage, String name) async {
   return storage.bucket(name);
 }
 
-Future deleteFromBucket(Bucket bucket, String objectName) async {
+/// Deletes a single object from the [bucket].
+///
+/// Returns `true` if the object was deleted by this operation, `false` if it
+/// didn't exist at the time of the operation.
+Future<bool> deleteFromBucket(Bucket bucket, String objectName) async {
   try {
     await bucket.delete(objectName);
+    return true;
   } on DetailedApiRequestError catch (e) {
     if (e.status != 404) {
       rethrow;
     }
+    return false;
+  }
+}
+
+/// Deletes a [folder] in a [bucket], recursively listing all of its subfolders.
+///
+/// Returns the number of objects deleted.
+Future<int> deleteBucketFolderRecursively(
+  Bucket bucket,
+  String folder, {
+  int concurrency,
+}) async {
+  if (!folder.endsWith('/')) {
+    throw ArgumentError('Folder path must end with `/`: "$folder"');
+  }
+  final deleter = _ObjectDeleter(bucket, concurrency);
+  final folders = <String>[folder];
+  while (folders.isNotEmpty) {
+    final currentFolder = folders.removeLast();
+    await for (final obj in bucket.list(prefix: currentFolder)) {
```

Comment:
Hmm, wouldn't it be better to implement something on `Bucket` that supports us in doing this?
Maybe `Bucket.list(prefix: ..., delimiter: '')`

Basically just implement a feature that let's users pass in a delimiter, and then make the empty delimiter special (so that it means "no delimiter")

Reply:
Maybe a `Bucket.list` with `recursive: true` would help. I'd find it confusing to set the delimiter to empty in order to get recursive results.